### PR TITLE
Add ACesiumCameraManager::RemoveCamera

### LINKED ISSUE #1056: Add ACesiumCameraManager::RemoveCamera
The camera manager currently allows users to add new camers, and update existing ones. But it doesn't allow an old camera to be removed. We should add a `RemoveCamera` method.

### PR DESCRIPTION
**Pull request by csciguy8**
Add null pointer check in ACesiumCameraManager::GetDefaultCameraManager.

Add new unit tests for CesiumCameraManager.

Closes #1056

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @csciguy8!

* :x: Missing CLA.
    * Please send in a [Contributor License Agreement](https://github.com/CesiumGS/cesium/blob/master/CONTRIBUTING.md#contributor-license-agreement-cla) (CLA) and comment back here to let us know to check this!
    * Maintainers, [here's how to check](https://github.com/CesiumGS/cesium/blob/master/Documentation/Contributors/CLAs/README.md) for a submitted CLA.

Reviewers, don't forget to make sure that:

- [ ] [Cesium for Unreal Samples](https://github.com/CesiumGS/cesium-unreal-samples) works.

**Comment by csciguy8** [comment]
@kring, tagging you as a reviewer since this relates to the new Unreal unit tests that you set up (the branch I started from).

A couple notes about the new camera manager tests:

- I named the new test file with the extension `.spec.cpp`, from Unreal's recommendations for Behavior Driven Design, [link](https://docs.unrealengine.com/4.27/en-US/TestingAndOptimization/Automation/AutomationSpec/)
- This differs from your first test, which has the TestXXXXXX.cpp naming scheme
- The new tests in CesiumCameraManagerSpec assume there's always a world context available from the global GEngine. Is this a reasonable based on the way we plan to run these?

**Comment by csciguy8** [comment]
@kring @j9liu In general, I'm happy with these changes, but do have a comment about design.

From what I can tell of the implementation, ACesiumCameraManager does these two things:
- Makes sure there's only one instance of itself persistent in an active level
- Serves as a wrapper for a container of FCesiumCamera instances

My first thought when reviewing this class, is it even needed? What logic is camera specific here? Could this be a generalized (templated) class that could contain anything? And if so, what is Cesium-specific here? Can we accomplish this functionality by using built-in Unreal techniques?

No action needed, but just food for thought for future code design discussions.

File: Source/CesiumRuntime/Private/Tests/CesiumCameraManager.spec.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#include "CesiumCameraManager.h"
+#include "Engine/Engine.h"
+#include "Engine/World.h"
+#include "Misc/AutomationTest.h"
+
+BEGIN_DEFINE_SPEC(
+    CesiumCameraManagerSpec,
```
Comment (j9liu) [inline Source/CesiumRuntime/Private/Tests/CesiumCameraManager.spec.cpp:7]: Just a drive-by comment, since I was looking at this to write my own unit tests.

In Unreal, structs are often prefixed with F, like `FStructName`. I looked at an example spec `AutomationDriver.spec.cpp` and it also defines the spec struct with an F prefix. So let's add it here for consistency.

```suggestion
    FCesiumCameraManagerSpec,
```

Also, can the naming scheme be updated for `TestGeoTransforms.cpp` as well?

**Comment by kring** [comment]
Thanks @csciguy8!

> I named the new test file with the extension .spec.cpp, from Unreal's recommendations for Behavior Driven Design, [link](https://docs.unrealengine.com/4.27/en-US/TestingAndOptimization/Automation/AutomationSpec/)

👍 

> The new tests in CesiumCameraManagerSpec assume there's always a world context available from the global GEngine. Is this reasonable based on the way we plan to run these?

I'm not sure. I guess we'll find out when we go to try to run these on our CI system. Surely it'll be possible to get a UWorld somehow, though, so at worst we'll have to make minor changes to this. It looks fine for now.

> Could this be a generalized (templated) class that could contain anything?

I don't think it's possible to create templatized classes that are accessible from Blueprints, but I could be wrong. In any case we have a lot of duplicated "find or create the thing of this type in the level" in a few different classes, so factoring that out, at least, would be a nice improvement.

Couple other small things, please:

1. [ ] Update CHANGES.md.
2. [ ] Change the original issue to say `Fixes #1056` (or `Closes #1056` if you prefer). That way GitHub will automatically close that issue when this PR is merged.

File: Source/CesiumRuntime/Private/Tests/CesiumCameraManager.spec.cpp (hunk 1 above)
Comment (csciguy8) [reply Source/CesiumRuntime/Private/Tests/CesiumCameraManager.spec.cpp]: Committed changes to address both points. Thanks!

**Comment by csciguy8** [comment]
Committed changes to address review points. Thanks @kring @j9liu

**Comment by csciguy8** [comment]
Pinging again for a merge @kring. Looks reviewed and all checks have passed.

**Comment by kring** [comment]
Thanks for the reminder @csciguy8, and for the PR!
